src/common/types.cpp

```diff
@@ -1258,42 +1285,47 @@ PhysicalType EnumType::GetPhysicalType(idx_t size) {
 void ExtraTypeInfo::Serialize(ExtraTypeInfo *info, FieldWriter &writer) {
 	if (!info) {
 		writer.WriteField<ExtraTypeInfoType>(ExtraTypeInfoType::INVALID_TYPE_INFO);
+		writer.WriteString(string());
 	} else {
 		writer.WriteField<ExtraTypeInfoType>(info->type);
+		writer.WriteString(info->alias);
 		info->Serialize(writer);
 	}
 }
 shared_ptr<ExtraTypeInfo> ExtraTypeInfo::Deserialize(FieldReader &reader) {
 	auto type = reader.ReadRequired<ExtraTypeInfoType>();
+	auto alias = reader.ReadField<string>(string());
 	switch (type) {
 	case ExtraTypeInfoType::INVALID_TYPE_INFO:
 		return nullptr;
+	case ExtraTypeInfoType::GENERIC_TYPE_INFO:
+		return make_shared<ExtraTypeInfo>(type, alias);
 	case ExtraTypeInfoType::DECIMAL_TYPE_INFO:
-		return DecimalTypeInfo::Deserialize(reader);
+		return DecimalTypeInfo::Deserialize(reader, alias);
```

Comment:
Rather than expanding all of these functions with a new parameter, could we deserialize first and then assign the alias afterwards (if there is any)?